src/algorithm/native/to_wkt.rs

```diff
@@ -376,17 +376,130 @@ fn add_coords<W: Write, C: CoordTrait<T = f64>>(
     writer: &mut W,
     mut coords: impl Iterator<Item = C>,
 ) -> Result<(), Error> {
-    writer.write_str("(")?;
+    writer.write_char('(')?;
 
     let first = coords.next().unwrap();
     add_coord(writer, first)?;
 
     for coord in coords {
-        writer.write_str(",")?;
+        writer.write_char(',')?;
         add_coord(writer, coord)?;
     }
 
-    writer.write_str(")")?;
+    writer.write_char(')')?;
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use arrow_buffer::OffsetBuffer;
+
+    use super::*;
+    use crate::{
+        array::{CoordBuffer, InterleavedCoordBuffer},
+        scalar::{
+            OwnedLineString, OwnedMultiLineString, OwnedMultiPoint, OwnedMultiPolygon, OwnedPoint,
+            OwnedPolygon,
+        },
+    };
+
+    #[test]
+    fn point() {
+        let coords = InterleavedCoordBuffer::<2>::new(vec![1., 2.].into());
+        let point = OwnedPoint::new(CoordBuffer::Interleaved(coords), 0);
+
+        let mut wkt = String::new();
+        point_to_wkt(&point, &mut wkt).unwrap();
+
+        assert_eq!(&wkt, "POINT (1.0 2.0)");
+    }
+
+    #[test]
+    fn linestring() {
+        let coords = InterleavedCoordBuffer::<2>::new(vec![1., 2., 3., 4., 5., 6.].into());
+        let linestring = OwnedLineString::new(
+            CoordBuffer::Interleaved(coords),
+            OffsetBuffer::<i32>::new(vec![0, 3].into()),
+            0,
+        );
+
+        let mut wkt = String::new();
+        linestring_to_wkt(&linestring, &mut wkt).unwrap();
+
+        assert_eq!(&wkt, "LINESTRING (1.0 2.0,3.0 4.0,5.0 6.0)");
```

Comment:
I don't think it's enough to assert that `wkt` matches this string because it's not obvious at a glance that this string is actually valid WKT.